### PR TITLE
Added an example of Liskov Substitution Principle.

### PR DESCRIPTION
**Pull request by atafliovich**
Added a "classic" example of Liskov Substitution Principle: "Square is not a Rectangle".

### CODE REVIEW

**Review by thirug13 — CHANGES_REQUESTED** [review]
The code does not pass the checkstyle.

**Comment by atafliovich** [comment]
Thank you! I think I've fixed the issues.

File: lsp/Main.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+  /**
+   * Run a demo.
+   * @param args The ususal.
+   */
+  public static void main(String[] args) {
+    Rectangle r = getRectangle();
+    r.setHeight(10);
+    r.setWidth(50);
+    System.out.println(r.area()); // it's not a rectangle!
```
Comment (thirug13) [inline lsp/Main.java]: Could you also include an explanatory print message in Main? This will benefit the students who miss your explanation in class.

**Review by thirug13 — CHANGES_REQUESTED** [review]

File: lsp/Main.java (hunk 1 above)
Comment (atafliovich) [reply lsp/Main.java]: Good idea! I added some print statements.

File: lsp/Main.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package lsp;
+
+public class Main {
+
+  /**
+   * Run a demo.
+   * @param args The ususal.
+   */
+  public static void main(String[] args) {
+    Rectangle r = getRectangle();
```
Comment (thirug13) [inline lsp/Main.java]: Could you please rename the "r" variable so that the name is meaningful? We want the code to be readable.

**Review by thirug13 — CHANGES_REQUESTED** [review]

File: lsp/Main.java (hunk 2 above)
Comment (atafliovich) [reply lsp/Main.java]: Good suggestion. Done!

**Review by thirug13 — APPROVED** [review]
Code reviewed, looks good.

**Pull request merged**